app/scripts/controllers/plugins.js

```diff
@@ -0,0 +1,149 @@
+const ObservableStore = require('obs-store')
+const EventEmitter = require('safe-event-emitter')
+const extend = require('xtend')
+const SES = require('ses')
+const Box = require('3box/dist/3box.min')
+
+class PluginsController extends EventEmitter {
+
+  constructor (opts = {}) {
+    super()
+    const initState = extend({
+      plugins: {},
+      pluginStates: {},
+    }, opts.initState)
+    this.store = new ObservableStore(initState)
+
+    this.provider = opts.provider
+    console.log('!!!!! opts', opts)
+    this._onUnlock = opts._onUnlock
+    this._onNewTx = opts._onNewTx
+    this._subscribeToPreferencesControllerChanges = opts._subscribeToPreferencesControllerChanges
+    this._updatePreferencesControllerState = opts._updatePreferencesControllerState
+    this._signPersonalMessage = opts._signPersonalMessage
+    this._getAccounts = opts._getAccounts
```

Comment:
We could probably simplify some of this by simply passing in the full `metamaskController.getApi()` result into the `plugins` controller, so it has access to all functions our UI has, since we may eventually pass any/all of that in here.